### PR TITLE
Add an error code for when the path fails

### PR DESCRIPTION
**Pull request by martinthomson**
This is a bit of a niche code, as it is only narrowly useful for
signaling loss of paths.  But people wanted it and it doesn't code that
much.

Note the target of this pull request.

For #4257.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -2311,7 +2311,9 @@ When an endpoint abandons path validation, it determines that the path is
 unusable.  This does not necessarily imply a failure of the connection -
 endpoints can continue sending packets over other paths as appropriate.  If no
 paths are available, an endpoint can wait for a new path to become available or
-close the connection.
+close the connection.  An endpoint that has no valid network path to its peer
+MAY signal this using the NO_VIABLE_PATH connection error, noting that this is
+only possible if the network has a limited MTU.
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:2316]:
```suggestion
only possible if the network path exists but does not support the required
MTU {{datagram-size}}.
```

**Review by janaiyengar — APPROVED** [review]
one suggestion, but lgtm

**Comment by martinthomson** [comment]
I'm going to leave this for a bit, but if people like the idea, I plan to merge it into #4264.

**Review by kazuho — APPROVED** [review]
Thank you for the PR.

This error code will be useful, as the only other way of closing the connection would be to use NO_ERROR. Then, it would be impossible to know why the connection was abruptly closed.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -6411,6 +6414,12 @@ AEAD_LIMIT_REACHED (0xf):
 : An endpoint has reached the confidentiality or integrity limit for the AEAD
   algorithm used by the given connection.
 
+NO_VIABLE_PATH (0x10):
+
+: An endpoint has determined that the network path is incapable of supporting
+  QUIC.  An endpoint is unlikely to receive CONNECTION_CLOSE carrying this code
+  unless the reason is to do the path not supporting a large enough MTU.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:6421]:
```suggestion
  except when the path does not support a large enough MTU.
```

**Review by MikeBishop — APPROVED** [review]
Yes, there's value in having a way to fail fast.

**Pull request merged**